libsupport/include/katana/EntityTypeManager.h

```diff
@@ -230,6 +231,25 @@ class KATANA_EXPORT EntityTypeManager {
   /// if it doesn't exist.
   Result<EntityTypeID> GetOrAddEntityTypeID(const std::string& name);
 
+  katana::TypeNameSet EntityTypeToTypeNameSet(
+      katana::EntityTypeID type_id) const {
+    katana::TypeNameSet type_name_set;
+    if (type_id == katana::kUnknownEntityType ||
+        type_id == katana::kInvalidEntityType ||
+        type_id >= GetNumEntityTypes()) {
+      return type_name_set;
+    }
```

Comment:
These should be errors or (in the case of Unknown and maybe Invalid) return the obvious name. Returning and empty set it a silent failure.

Reply:
Fair enough.  My feeling is that Unkown is a valid type that should be represented by the empty set.  I will return errors in the other cases.

Reply:
You are right. Unknown should be the empty set (because it represents the lack of any known type). The others should be errors as you say. Good thinking.